src/menu/cutsceneview.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * This software is licensed under the terms of the MIT License.
+ * See COPYING for further information.
+ * ---
+ * Copyright (c) 2011-2019, Lukas Weber <laochailan@web.de>.
+ * Copyright (c) 2012-2019, Andrei Alexeyev <akari@taisei-project.org>.
+ */
+
+#include "taisei.h"
+
+#include "eventloop/eventloop.h"
+#include "menu/menu.h"
+#include "cutsceneview.h"
+#include "common.h"
+#include "options.h"
+#include "global.h"
+#include "video.h"
+#include "cutscenes/cutscene.h"
+#include "cutscenes/scenes.h"
+#include "progress.h"
+
+static void draw_cutsceneview_menu(MenuData *m) {
+	draw_options_menu_bg(m);
+	draw_menu_title(m, "Cutscene Viewer");
+	draw_menu_list(m, 100, 100, NULL, SCREEN_H);
+}
+
+static void restart_menu_bgm(CallChainResult ccr) {
+	audio_bgm_play(res_bgm("menu"), true, 0, 0);
+}
+
+static void cutscene_player(MenuData *m, void *arg) {
+	CutsceneID cutscene_id = (uintptr_t)arg;
+	cutscene_enter(CALLCHAIN(restart_menu_bgm, NULL), cutscene_id);
+}
+
+MenuData *create_cutsceneview_menu(void) {
+	MenuData *m = alloc_menu();
+
+	m->draw = draw_cutsceneview_menu;
+	m->logic = animate_menu_list;
+	m->flags = MF_Abortable;
+	m->transition = TransFadeBlack;
+
+	for(uintptr_t id = 0; id < NUM_CUTSCENE_IDS; id++) {
+		if(progress_is_cutscene_unlocked(id)) {
+			const Cutscene *cs = g_cutscenes + id;
+			add_menu_entry(m, cs->name, cutscene_player, (void*)id);
```

Comment:
Rather than doing this, consider adding a `const char *cutscene_get_name(CutsceneID id)` API to `cutscenes/cutscene.h`.

`cutscenes/scenes.h` is kind of a private header that leaks a lot of implementation details into the namespace.